cli/src/pcluster/utils.py

```diff
@@ -117,14 +119,22 @@ def update_stack_template(stack_name, updated_template, cfn_parameters):
 
 
 def get_region():
-    """Get AWS_DEFAULT_REGION from the environment."""
-    return os.environ.get("AWS_DEFAULT_REGION")
+    """
+    Get region used internally for all the AWS calls.
+
+    The region from the env has higher priority because it can be explicitly set from the code (e.g. unit test).
+    """
+    region = os.environ.get("AWS_DEFAULT_REGION")
+    if not region:
+        bot3_session = boto3.session.Session()
+        region = bot3_session.region_name
+
+    return region
```

Comment:
Minor: why not return `boto3.session.Session().region_name` directly?

Reply:
Fixed